sdk/src/main/java/com/hedera/hashgraph/sdk/Mnemonic.java

```diff
@@ -557,28 +559,101 @@ public PrivateKey toStandardEd25519PrivateKey(String passphrase, int index) {
         return derivedKey;
     }
 
+    /**
+     * Converts a derivation path from string to an array of integers.
+     * Note that this expects precisely 5 components in the derivation path,
+     * as per BIP-44:
+     * `m / purpose' / coin_type' / account' / change / address_index`
+     * Takes into account `'` for hardening as per BIP-32,
+     * and does not prescribe which components should be hardened.
+     *
+     * @param derivationPath  the derivation path in BIP-44 format,
+     *                        e.g. "m/44'/60'/0'/0/0"
+     * @return an array of integers designed to be used with PrivateKey#derive
+     */
+    private int[] calculateDerivationPathValues(String derivationPath) {
```

Comment:
Could benefit from additional validation and exception handling to cover invalid input formats
 
```java
    private int[] calculateDerivationPathValues(String derivationPath) throws IllegalArgumentException {
        if (derivationPath == null || derivationPath.isEmpty()) {
            throw new IllegalArgumentException("Derivation path cannot be null or empty");
        }

        Pattern pattern = Pattern.compile("m/(\\d+'?)/(\\d+'?)/(\\d+'?)/(\\d+'?)/(\\d+'?)");
        Matcher matcher = pattern.matcher(derivationPath);

        if (!matcher.matches()) {
            throw new IllegalArgumentException("Invalid derivation path format. Expected format: m/44'/60'/0'/0/0");
        }

        int[] numbers = new int[5];
        boolean[] isHardened = new boolean[5];

        try {
            // Extract numbers and determine if hardened
            for (int i = 1; i <= 5; i++) {
                String value = matcher.group(i);
                if (value.endsWith("'")) {
                    isHardened[i - 1] = true;
                    value = value.substring(0, value.length() - 1);
                } else {
                    isHardened[i - 1] = false;
                }
                numbers[i - 1] = Integer.parseInt(value);
            }
        } catch (NumberFormatException e) {
            throw new IllegalArgumentException("Invalid number format in derivation path", e);
        }

        int[] values = new int[5];
        for (int i = 0; i < numbers.length; i++) {
            values[i] = (isHardened[i] ? Bip32Utils.toHardenedIndex(numbers[i]) : numbers[i]);
        }

        return values;
    }

```

Reply:
thanks for the suggestion - will look into this!

Reply:
@0xivanov I have implemented this in [8f7f0e6](https://github.com/hashgraph/hedera-sdk-java/pull/1842/commits/8f7f0e6d8ec19a13ae08b3edc15d941ae3378905)

Note that I'm not sure if the `NumberFormatException` scenario can occur, thanks to the Regex that would catch it upon executing `matcher.matches()`.
Do you prefer if I leave that check in anyway?
Or should I remove it from the code?
Or is there something I'm missing (something that could slip past the Regex pattern match test)?

Reply:
You are right that `NumberFormatException` is unlikely to occur, however, for defensive programming and to ensure robustness, it's still a good practice to handle any unexpected exceptions that might arise.

Reply:
Ok got it.
I will leave it in then.
But I won't have test coverage for that `catch` - I assume that this is OK.